gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiReactorHandlerFactory.java

```diff
@@ -217,14 +208,34 @@ public ReactorHandler create(Api api) {
         return null;
     }
 
-    protected ExecutionContextFactory executionContextFactory(ComponentProvider componentProvider) {
-        return new ExecutionContextFactory(componentProvider);
+    private boolean isV3ExecutionMode(Api api) {
+        return api.getExecutionMode() == null || api.getExecutionMode() == ExecutionMode.V3;
+    }
+
+    protected <T extends AbstractExecutionContextFactory> T executionContextFactory(
+        Api api,
+        ComponentProvider componentProvider,
+        DefaultReferenceRegister referenceRegister
+    ) {
+        final AbstractExecutionContextFactory executionContextFactory;
```

Comment:
Could define a interface to avoid avoid this Abstract variable and also to clean the generic cast at the end which is unchecked?

Reply:
Yes sounds to be a better approach. Need to think about how to deal with a factory able to create v3 `ExecutionContext` and Jupiter `ExecutionContext` in the same time 🧐

Reply:
Finally, no interface was require as a different implementation has been recreated.